algopy_testing/tests/test_miscellaneous_op.py

```diff
@@ -0,0 +1,1581 @@
+import base64
+import math
+import re
+from pathlib import Path
+
+import algokit_utils
+import pytest
+from algokit_utils import ApplicationClient, get_localnet_default_account
+from algokit_utils.config import config
+from algopy import BigUInt, UInt64, op
+from algopy_testing.constants import MAX_BYTES_SIZE, MAX_UINT64, MAX_UINT512
+from algosdk.error import ABIEncodingError
+from algosdk.v2client.algod import AlgodClient
+from algosdk.v2client.indexer import IndexerClient
+
+from tests.common import AVMInvoker, create_avm_invoker
+from tests.util import get_sha256_hash, int_to_bytes
+
+ARTIFACTS_DIR = Path(__file__).parent / "artifacts"
+APP_SPEC = ARTIFACTS_DIR / "MiscellaneousOps" / "data" / "MiscellaneousOpsContract.arc32.json"
+
+
+def _too_big_error(max_length: int) -> str:
+    return re.escape(f"expected value <= {max_length}")
+
+
+_too_big64_error = _too_big_error(MAX_UINT64)
+_too_big512_error = _too_big_error(MAX_UINT512)
+
+_avm_int_arg_overflow_error = "is not a non-negative int or is too big to fit in size"
+_avm_bytes_arg_overflow_error = "math attempted on large byte-array"
+
+_extract_out_of_bound_error = re.compile("extraction (start|end) \\d+ is beyond length")
+
+
+@pytest.fixture(scope="session")
+def ops_client(algod_client: AlgodClient, indexer_client: IndexerClient) -> ApplicationClient:
+    config.configure(
+        debug=True,
+    )
+
+    client = ApplicationClient(
+        algod_client,
+        APP_SPEC,
+        creator=get_localnet_default_account(algod_client),
+        indexer_client=indexer_client,
+    )
+
+    client.deploy(
+        on_schema_break=algokit_utils.OnSchemaBreak.AppendApp,
+        on_update=algokit_utils.OnUpdate.AppendApp,
+    )
+
+    return client
+
+
+@pytest.fixture(scope="module")
+def get_ops_avm_result(
+    ops_client: ApplicationClient,
+) -> AVMInvoker:
+    return create_avm_invoker(ops_client)
+
+
+@pytest.mark.parametrize(
+    ("a", "b"),
+    [
+        (0, 0),
+        (0, MAX_UINT64),
+        (MAX_UINT64, 0),
+        (1, 0),
+        (0, 1),
+        (100, 42),
+        (1, MAX_UINT64 - 1),
+        (MAX_UINT64 - 1, 1),
+        (100, MAX_UINT64),
+        (MAX_UINT64, MAX_UINT64),
+    ],
+)
+def test_addw(get_ops_avm_result: AVMInvoker, a: int, b: int) -> None:
+    avm_result = get_ops_avm_result("verify_addw", a=a, b=b)
+    avm_result_tuples = _bytes_to_uint64_tuple(avm_result)
+    result = op.addw(a, b)
+    assert avm_result_tuples == result
+
+
+@pytest.mark.parametrize(
+    ("a", "b"),
+    [(1, MAX_UINT64 + 1), (MAX_UINT64 + 1, 1), (0, MAX_UINT512), (MAX_UINT512 * 2, 0)],
+)
+def test_addw_input_overflow(get_ops_avm_result: AVMInvoker, a: int, b: int) -> None:
+    with pytest.raises(ABIEncodingError, match=_avm_int_arg_overflow_error):
```

Comment:
This particular error is coming from the SDK, not the AVM so I think we can just remove it and just have the testing implementation check below

Reply:
All tests for ABIEncodingError have been removed in this commit 414af4e35198510db081551ed9370a944178090c.